examples/goldilocks_models/find_goldilocks_models.cc

```diff
@@ -263,15 +272,26 @@ void getInitFileName(string * init_file, const string & nominal_traj_init_file,
   } else if (step_size_shrinked_last_loop && n_rerun == 0) {
     // the step size was shrink in previous iter and it's not a local rerun
     // (n_rerun == 0)
-    *init_file = to_string(iter-1) + "_" + to_string(sample) + string("_w.csv");
+    if (is_use_interpolated_initial_guess){
+        //      modified by Jianshu to test new initial guess
+        *init_file = set_initial_guess(dir, iter, sample, total_sample_num, min_sl, max_sl, min_gi, max_gi,
+                min_tr, max_tr);
+    }
+    else{
+        *init_file = to_string(iter-1) + "_" + to_string(sample) + string("_w.csv");
+    }
   } else if (sample_idx_to_help >= 0) {
     *init_file = to_string(iter) + "_" + to_string(sample_idx_to_help) +
                  string("_w.csv");
   } else if (rerun_current_iteration) {
     *init_file = to_string(iter) + "_" + to_string(sample) + string("_w.csv");
-  } else {
-    *init_file = to_string(iter - 1) +  "_" +
-                 to_string(sample) + string("_w.csv");
+  }else if(is_use_interpolated_initial_guess){
+//      modified by Jianshu to test new initial guess
+      *init_file = set_initial_guess(dir, iter, sample, total_sample_num, min_sl, max_sl, min_gi, max_gi,
+              min_tr, max_tr);
+  } else{
+      *init_file = to_string(iter - 1) +  "_" +
+                   to_string(sample) + string("_w.csv");
```

Comment:
It's a little bit clearer to write
} else{
    if (is_use_interpolated_initial_guess) {
        //      modified by Jianshu to test new initial guess
        *init_file = set_initial_guess(dir, iter, sample, total_sample_num, min_sl, max_sl, min_gi, max_gi,
              min_tr, max_tr);
    } else {
        *init_file = to_string(iter - 1) +  "_" +
                   to_string(sample) + string("_w.csv");
    }
}

Reply:
Resolved